ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/Carrier.java

```diff
@@ -151,5 +167,62 @@ public void setEnemySpots(ArrayList<EnemySpot> enemySpots) {
     public HealthMessenger getHealthMessenger() {
         return healthMessenger;
     }
+    
+    /**
+     * Executes an attack using a player's position and direction of attack.
+     * @param direction
+     * 			the direction of the attack (String)
+     */
+    public void handleAttack(String direction) {
+        switch (direction) {
+            case "left":
+                if (side.equals("left")) {
+                    attack(2);
+                }
+                else {
+                    attack(0);
+                }
+                break;
+            case "middle":
+                attack(1);
+                break;
+            case "right":
+                if (side.equals("left")) {
+                    attack(0);
+                }
+                else {
+                    attack(1);
+                }
+                break;
+            default:
+                throw new IllegalArgumentException();
+        }
+    }
+    
+    /**
+     * This function checks whether or not the player hits an enemy and
+     * performs an attack when it does.
+     * It also sends a HitMissMessage to the player.
+     * @param direction 
+     *              the direction of the attack, represented by an integer
+     *              (see function handleAttack)
+     */
+    public void attack(int direction) {
```

Comment:
That is because I had to make it work with damian's work. handleAttack converts the direction string into the right integer that can be used with his ArrayList of EnemySpots.